### DISCUSSION TITLE
[ZK Stack] - prover fails to compile

### DISCUSSION THREAD

**Discussion by qiwenzhou** [zkSync-Community-Hub/zksync-developers#536] [ZK Stack]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I was running ./setup gpu under /prover. Got error like:
<img width="913" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/fb05c65b-7071-4bda-b4bd-f78f7e2e7bd3">
I tried in another server and it worked.
But I tried to fix this problem.It looks like a rust problem.Any idea about how to fix this?

### Expected Behavior

./setup gpu finished compile

### Current Behavior

cannot finish compiling

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by qiwenzhou** [comment]
got errors like this
<img width="854" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/fdc0c587-093b-406b-8ccd-b4ccb76e93dd">
<img width="695" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/1f567e74-a4f5-42f0-8ce3-759ed0990e07">
<img width="829" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/cf0dac3a-a79e-4d63-bdeb-5559b24028cb">

**Reply by bxpana** [reply]
> I tried in another server and it worked.
But I tried to fix this problem.It looks like a rust problem.Any idea about how to fix this?

Could you further explain what you mean by you tried in another server and it worked?

What makes you think it's a rust problem?

**Reply by qiwenzhou** [reply]
Sorry for late reply. I deployed another hyperchain and worked. I ran rustc update before this error and I guess this could be the reason.
